services/datalake/pod-datalake/src/datalake/db.ts

```diff
@@ -284,6 +294,34 @@ export class PostgresDB implements BlobDB {
     )
   }
 
+  async getMeta (ctx: MeasureContext, blobId: BlobId): Promise<BlobMeta | null> {
+    const { workspace, name } = blobId
+
+    const rows = await this.db.execute<BlobMetaRecord[]>(
+      `
+      SELECT m.workspace, m.name, m.meta
+      FROM blob.blob b
+      LEFT JOIN blob.meta as m ON m.workspace = b.workspace AND m.name = b.name
+      WHERE b.workspace = $1 AND b.name = $2 and b.deleted_at IS NULL
+    `,
+      [workspace, name]
+    )
+
+    return rows.length > 0 ? rows[0].meta ?? {} : null
+  }
+
+  async setMeta (ctx: MeasureContext, blobId: BlobId, meta: BlobMeta): Promise<void> {
+    const { workspace, name } = blobId
+
+    await this.db.execute(
+      `
+      UPSERT INTO blob.meta (workspace, name, meta)
```

Comment:
Looks like `UPSERT` command exists in CockroachDB, but in standard PostgreSQL there is only INSERT ... ON CONFLICT
Wouldn't it better to use [INSERT ... ON CONFLICT](https://www.cockroachlabs.com/docs/v25.1/insert#on-conflict-clause) statement?

Reply:
Can be done separately if we decide we need postgresql compatibility.